Add explanation about @critical_section target directive (#1246)
python/devguide: development-tools/clinic.rst

      return return_value;
   }

> You can lock up to two additional objects

Reviewer: ```suggestion
You can lock one or two additional objects
```
Maybe this is clearer since I initially read it as "(lock up) to" instead of "lock (up to)".

Reviewer 2: Thanks, that's less confusing!

by supplying their C variable names as arguments
to the ``@critical_section`` directive.
This example from :cpy-file:`Modules/_weakref.c` takes
one additional argument (a C variable named ``object``)::

   /*[clinic input]
   @critical_section object
   _weakref.getweakrefcount -> Py_ssize_t

>    object: object

Reviewer: Maybe it would be better to find an example that uses a different name, in order to avoid confusion between the variable name and its type.

Reviewer 2: Well, it does not have to be a real example :) The Argument Clinic docs are full of code examples that are just made up.

      return return_value;
   }

> If you want to specify a target object to be locked then you can use a target directive for
> ``@critical_section``. Since the ``@critical_section`` locks the object which is located
> as the first argument implicitly, you will need this feature when the ``@critical_section``
> locks an unwanted target object. Note that explicit target declaration only supports up to 2 objects.
> 
> Example from :cpy-file:`Modules/_weakref.c`::

Reviewer 2: I wonder if this text is overly complex. How about reducing it to something like this?

```suggestion
You can lock up to two additional objects
by supplying their C variable names as arguments
to the ``@critical_section`` directive.
This example from :cpy-file:`Modules/_weakref.c` takes
one additional argument (a C variable named ``object``)::
```

Author: Good!
